reference/test/solver/batch_bicgstab_kernels.cpp

```diff
@@ -105,8 +105,13 @@ TYPED_TEST(BatchBicgstab, StencilSystemLoggerLogsResidual)
         ASSERT_LE(
             res_log_array[i] / this->linear_system.host_rhs_norm->at(i, 0, 0),
             this->solver_settings.residual_tol);
-        ASSERT_NEAR(res_log_array[i], res.host_res_norm->get_const_values()[i],
-                    10 * this->eps);
+        if (!std::is_same<real_type, gko::half>::value) {
+            // There is no guarantee of this condition. We disable this check in
+            // half.
```

Comment:
What do you mean there is no guarantee for this condition ? I would assume that they are atleast close.

Reply:
the implicit residual norm can be quite different from the actual residual norm after many iterations.

Reply:
IMO the check can be removed completely. We can only guarantee that the (implicit) residual will be less than the given tolerance, which is checked above. So this is a quite unrelated check.

Reply:
I would still suggest to remove this check for all value types.